internal/ias/client.go

```
@@ -27,74 +29,172 @@ func NewIasClient(iasTenantUrl, user, password string) (Client, error) {
+	if res.StatusCode() != http.StatusCreated {
+		ctrl.Log.Error(err, "Failed to create application", "name", name, "statusCode", res.StatusCode())
+		return uuid.UUID{}, errors.New("failed to create application")
+	}
+
+	return extractApplicationId(res)
+}
+
+func (c client) createSecret(ctx context.Context, appId uuid.UUID) (string, error) {
+	res, err := c.api.CreateApiSecretWithResponse(ctx, appId, newSecretRequest())
+	if err != nil {
+		return "", err
+	}
+
+	if res.StatusCode() != http.StatusCreated {
+		ctrl.Log.Error(err, "Failed to create api secret", "id", appId, "statusCode", res.StatusCode())
+		return "", errors.New("failed to create api secret")
+	}
+
+	return *res.JSON201.Secret, nil
+}
+
+func (c client) getClientId(ctx context.Context, appId uuid.UUID) (string, error) {
 	// The client ID is generated only after an API secret is created, so we need to retrieve the application again to get the client ID.
 	applicationResponse, err := c.api.GetApplicationWithResponse(ctx, appId, &api.GetApplicationParams{})
 	if err != nil {
-		return Application{}, err
+		return "", err
 	}
 
 	if applicationResponse.StatusCode() != http.StatusOK {
-		ctrl.Log.Error(err, "Failed to create api secret", "id", appId, "statusCode", createSecretResponse.StatusCode())
-		return Application{}, fmt.Errorf("failed to create api secret")
+		ctrl.Log.Error(err, "Failed to retrieve client ID", "id", appId, "statusCode", applicationResponse.StatusCode())
+		return "", errors.New("failed to retrieve client ID")
 	}
-	clientId := *applicationResponse.JSON200.UrnSapIdentityApplicationSchemasExtensionSci10Authentication.ClientId
+	return *applicationResponse.JSON200.UrnSapIdentityApplicationSchemasExtensionSci10Authentication.ClientId, nil
+}
 
-	return NewApplication(clientId, clientSecret, c.tenantUrl), nil
+func (c client) deleteApplication(ctx context.Context, id uuid.UUID) error {
+	res, err := c.api.DeleteApplicationWithResponse(ctx, id)
+	if err != nil {
```

it'd be better to ignore the error if the application is not found. I assume we get an error unless an app found.

---

I think we should handle the error since there is an application we want to delete. In case the API call `DeleteApplicationWithResponse` fails for some reason, e.g. connection is broken/dropped, we would need to propagate that the deletion didn't happen. 
I think what you mean is we should ignore the 404 that might be returned from the deletion.
I checked the [delete app API docs](https://api.sap.com/api/SCI_Application_Directory/path/deleteApplication) and 404 is note mentioned. Unfortunately the implementation behaves different as it returns 404. So I'd propose to add a handling to ignore 404, but keep the return of the err in place.